app/helpers/icons.py

```diff
@@ -1,28 +1,44 @@
-import os
-
-from app.icon_set import IconSet
-from app.settings import COLORABLE_ICON_SETS
-from app.settings import IMAGE_FOLDER
+def get_icon_set_template_url(base_url=''):
+    """
+    Generate and return a template URL to access icon_sets' metadata
+    """
+    return f"{base_url}/{{icon_set_name}}"
 
 
-def get_all_icon_sets():
-    icon_sets = []
-    for root, dirs, files in os.walk(IMAGE_FOLDER):
-        for icon_set_name in dirs:
-            icon_sets.append(get_icon_set(icon_set_name))
-    return icon_sets
+def get_icon_template_url(base_url='', with_color=True):
+    """
+    Generate and returns a URL template for icons
+    """
+    color_part = ""
+    if with_color:
+        color_part = "-{r},{g},{b}"
+    return f"{get_icon_set_template_url(base_url)}/icon/{{icon_name}}" \
+           f"@{{icon_scale}}{color_part}.png"
 
 
-def get_icon_set(icon_set_name):
+def generate_full_icon_url(
+    base_url, icon_set_name, icon_name, icon_scale='1x', colorized=True, red=255, green=0, blue=0
+):
     """
+    Generate a full URL to an icon's image
     Args:
-        icon_set_name (str): The name of the icon set we want
+        base_url (str): The base URL of the app, can be found by app.helpers.url.get_base_url when
+            running through a Flask request (won't work in Unit test though)
+        icon_set_name (str): The name of the icon set in which belongs the icon we want
+        icon_name (str): The name of the icon
+        icon_scale (str): The scale of image we want (possible values are 1x, 2x or 0.5x).
+            1x means 48x48px image.
+        colorized (bool): Tells if the URL should contain a colorized part (RGB part)
+        red (int): Red value for this icon (only taken into account if colorized is True)
+        green (int): Green value for this icon (only taken into account if colorized is True)
+        blue (int): Blue value for this icon (only taken into account if colorized is True)
     Returns:
-        (IconSet): the icon set if found, or None if no icon set with this name is found
+        A full blown URL to access this icon's image directly
     """
-    if icon_set_name:
-        icon_set = IconSet(icon_set_name, icon_set_name in COLORABLE_ICON_SETS)
-        # checking that this icon set really exists in the static/images folder
-        if icon_set.is_valid():
-            return icon_set
-    return None
+    return get_icon_template_url(base_url, colorized)\
+        .replace('{icon_set_name}', icon_set_name)\
+        .replace('{icon_name}', icon_name)\
+        .replace('{icon_scale}', icon_scale)\
+        .replace('{r}', f"{red}")\
+        .replace('{g}', f"{green}")\
+        .replace('{b}', f"{blue}")
```

Comment:
Couldn't you directly use `template_string.format(r=red, g=green, ...`? Then this function is probably not even needed and you can directly use `.format` to generate the concrete url...

Reply:
ok that's way clearer, I didn't know this `.format` function so I didn't use it 😛